lib/action_policy/i18n.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module ActionPolicy
+  module I18n # :nodoc:
+    class << self
+      def full_message(policy_class, rule)
+        unless defined?(::I18n)
+          raise "I18n is not available! Please make sure you have 'i18n' gem installed"
+        end
+
+        candidates = candidates_for(policy_class, rule)
+
+        ::I18n.t(
+          candidates.shift,
+          default: candidates,
+          scope: [:action_policy, :policy]
+        )
+      end
+
+      private
+
+      def candidates_for(policy_class, rule)
+        policy_hierarchy = policy_class.ancestors.select { |klass| klass < ActionPolicy::Base }
+        [
+          *policy_hierarchy.map { |klass| :"#{klass.identifier}.#{rule}" },
+          rule,
+          ::I18n.t(:unauthorized, scope: [:action_policy])
+        ]
+      end
+    end
+  end
+end
+
+I18n.load_path << Dir[File.expand_path("config/locales") + "/*.yml"] if defined?(::I18n)
```

Comment:
Maybe we can provide default message as a candidate?

I.e. passing `DEFAULT_UNAUTHORIZED_MESSAGE` as the last candidate.

When we add `::I18n.t(:unauthorized, scope: [:action_policy])` we assume that it's defined by the app.

Reply:
✔️